src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -0,0 +1,398 @@
+                                        (((uint)IPAddressToAdd & 0x0000FF00) << 8) |
+                                        (((uint)IPAddressToAdd >> 8) & 0x0000FF00) |
+                                        ((uint)IPAddressToAdd >> 24));
+#endif
+
+                GlobalLog.Print("currentArrayElement: " + currentArrayElement.ToString() + " nativePointer: " + nativePointer.ToString() + " IPAddressToAdd:" + IPAddressToAdd.ToString());
+
+                //
+                // ...and add it to the list
+                //
+                TempIPAddressList.Add(new IPAddress((long)IPAddressToAdd & 0x0FFFFFFFF));
+
+                //
+                // now get the next pointer in the array and start over
+                //
+                currentArrayElement = IntPtrHelper.Add(currentArrayElement, IntPtr.Size);
+                nativePointer = Marshal.ReadIntPtr(currentArrayElement);
+            }
+
+            HostEntry.AddressList = new IPAddress[TempIPAddressList.Count];
+            TempIPAddressList.CopyTo(HostEntry.AddressList, 0);
+
+            //
+            // Now do the same thing for the aliases.
+            //
+
+            var TempAliasList = new List<string>();
+
+            currentArrayElement = Host.h_aliases;
+            nativePointer = Marshal.ReadIntPtr(currentArrayElement);
+
+            while (nativePointer != IntPtr.Zero)
+            {
+                GlobalLog.Print("currentArrayElement: " + ((long)currentArrayElement).ToString() + "nativePointer: " + ((long)nativePointer).ToString());
+
+                //
+                // if it's not null it points to an Alias,
+                // read it...
+                //
+                AliasToAdd = Marshal.PtrToStringAnsi(nativePointer);
+
+                //
+                // ...and add it to the list
+                //
+                TempAliasList.Add(AliasToAdd);
+
+                //
+                // now get the next pointer in the array and start over
+                //
+                currentArrayElement = IntPtrHelper.Add(currentArrayElement, IntPtr.Size);
+                nativePointer = Marshal.ReadIntPtr(currentArrayElement);
+            }
+
+            HostEntry.Aliases = new string[TempAliasList.Count];
+            TempAliasList.CopyTo(HostEntry.Aliases, 0);
+
+            return HostEntry;
+        } // NativeToHostEntry
+
+        public static IPHostEntry GetHostByName(string hostName)
+        {
+            //
+            // IPv6 disabled: use gethostbyname() to obtain DNS information.
+            //
+            IntPtr nativePointer =
+                Interop.Winsock.gethostbyname(
+                    hostName);
+
+            if (nativePointer == IntPtr.Zero)
+            {
+                // This is for compatiblity with NT4/Win2k
+                // Need to do this first since if we wait the last error code might be overwritten.
+                SocketException socketException = new SocketException();
```

Comment:
Comments removed. The comment on 156 is an attempt to explain the code: calling the ctor for `SocketException()` without any parameter will call `Marshal.GetLastWin32Error()` and so preserve the last system error seen before other Win32 APIs are called.